src/maputils.erl

```diff
@@ -35,6 +36,18 @@ deep_get([Key|Keys], Map) ->
 deep_get([], Value) ->
     Value.
 
+deep_put([Key|Keys], Value, Map) when is_map(Map) ->
+    case maps:is_key(Key, Map) of
```

Comment:
If you use `maps:find/2` here instead, you don't have to do a `maps:get/2` later.